pkg/trace/writer/sender.go

```diff
@@ -32,7 +33,7 @@ func newSenders(cfg *config.AgentConfig, r eventRecorder, path string, climit, q
 	if e := cfg.Endpoints; len(e) == 0 || e[0].Host == "" || e[0].APIKey == "" {
 		panic(errors.New("config was not properly validated"))
 	}
-	client := cfg.HTTPClient()
+	client := httputils.NewClient(cfg.ConnectionResetInterval, cfg.HTTPClient)
```

Comment:
We should only use this client if there is `ConnectionResetInterval` set, otherwise there is no point to trigger all the new logic along with the mutex. This will also ensure that any previous (and default) functionality continues to work as it did in terms of performance.

Reply:
This could be reflected in the constructor name, `NewClientWithConnReset` for instance.

Reply:
I went with `NewResetClient`, and left the handling of a zero connection reset interval to the `ResetClient` so it's transparent to the caller (the overhead of the `ResetClient` with a disabled interval should be minimal: basically one read lock/read unlock and a pointer copy)